src/nvim/os/keyboard.c

```diff
@@ -0,0 +1,285 @@
+///                    in form of a pointer to integer bitmask.
+/// @param[out]  err  Location where error message is to be saved
+///
+/// @return -1 in case of error and a mask specifying which values
+///         in mods are valid otherwise.
+static ModMask os_get_locks_status(ModMask *const mods, Error *const err)
+{
+  *mods = 0;
+  if (GetKeyState(VK_CAPITAL) & 0x0001) {
+    *mods |= kCapsLock;
+  }
+  if (GetKeyState(VK_NUMLOCK) & 0x0001) {
+    *mods |= kNumLock;
+  }
+  if (GetKeyState(VK_SCROLL) & 0x0001) {
+    *mods |= kScrollLock;
+  }
+  return kCapsLock | kNumLock | kScrollLock;
+}
+
+#elif defined(__APPLE__)
+/// Get information about Num/Caps Lock state in MacOS
+///
+/// To be used in os_mods_status() function.
+///
+/// @param[out]  mods  Holds the information about the status of various locks
+///                    in form of a pointer to integer bitmask.
+/// @param[out]  err  Location where error message is to be saved
+///
+/// @return -1 in case of error and a mask specifying which values
+///         in mods are valid otherwise.
+static ModMask os_get_locks_status(ModMask *const mods, Error *const err)
+{
+  const CFMutableDictionaryRef mdict = IOServiceMatching(kIOHIDSystemClass);
+  const io_service_t ios = IOServiceGetMatchingService(kIOMasterPortDefault,
+                                                       (CFDictionaryRef)mdict);
+  if (!ios) {
+    if (mdict) {
+      CFRelease(mdict);
+    }
+    api_set_error(err, kErrorTypeException,
+                  "Unable to get service for default master port");
+    return -1;
+  }
+
+  io_connect_t ioc;
+  if (IOServiceOpen(ios, mach_task_self(), kIOHIDParamConnectType,
+                    &ioc) != kIOReturnSuccess) {
+    IOObjectRelease(ios);
+    api_set_error(err, kErrorTypeException,
+                  "Unable to get service for HID system class");
+    return -1;
+  }
+  IOObjectRelease(ios);
+
+  bool stateCaps;
+  if (IOHIDGetModifierLockState(ioc, kIOHIDCapsLockState,
+                                &stateCaps) != kIOReturnSuccess) {
+    IOServiceClose(ioc);
+    api_set_error(err, kErrorTypeException,
+                  "Unable to query CapsLock state");
+    return -1;
+  }
+
+  bool stateNums;
+  if (IOHIDGetModifierLockState(ioc, kIOHIDNumLockState,
+                                &stateNums) != kIOReturnSuccess) {
+    IOServiceClose(ioc);
+    api_set_error(err, kErrorTypeException,
+                  "Unable to query NumLock state");
+    return -1;
+  }
+  if (stateCaps) {
+    *mods |= kCapsLock;
+  }
+  if (stateNums) {
+    *mods |= kNumLock;
+  }
+  IOServiceClose(ioc);
+  return kCapsLock | kNumLock;
+}
+#else
+/// Dummy function used in case of no known platform detected.
+///
+/// @param[out]  mods  Holds the information about the status of various locks
+///                    in form of a pointer to integer bitmask.
+/// @param[out]  err  Location where error message is to be saved
+///
+/// @return Always return 0.
```

Comment:
It is supposed to be phrased like `@return` is the start of the sentence. E.g. in this case it would be `@return always 0.` (as you could not move “always” before `@return`). You may see in https://neovim.io/doc/dev/typval_8c.html#a2aeea7f4caab15609a6249a8d1e9a587 actually why it is so: `@return` starting block is rendered as a block which starts with an outstanding “Returns” header.